cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/ScenarioState.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.homeoffice.digital.sas.cucumberjparest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ScenarioState {
+
+  private String service;
+
+  public String trackService(String service) {
+    if (service != null && !service.isEmpty()) {
+      this.service = service;
+    } else {
```

Comment:
Do we need this `else `block? If the above `if` block is false wouldn't the service param be null/empty and then handled by the below assertion?

Reply:
The result of the else block is that when no service is provided it uses the last value provided i.e. this.service.
if service is set it updates the currently tracked state
if service is not set it uses the last tracked state
if service is not provided and has never previously been set it throws an error.
A new ScenarioState instance is create for each scenario

this allows
Scenario: ok
Given Trevor stores timecard in the timecard service
And Trevor gets timecard from the timecard service

Scenario: ok
Given Trevor stores timecard in the timecard service
And Trevor gets timecard 

Scenario: Fails
Given Trevor stores timecard
And Trevor gets timecard from the timecard service
because when the first step executes no service is specified and it's not been indicated in a previous step

Reply:
Ah right thanks for the clarification, that makes sense.
Not sure how far we intend to go with unit tests in this library, but it may be worth adding 3 simple tests to cover these conditions.